test/unit/org/apache/cassandra/index/sai/plan/QueryPlanRedactionTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.plan;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.Session;
+import org.apache.cassandra.index.sai.SAITester;
+import org.apache.cassandra.transport.ProtocolVersion;
+
+public class QueryPlanRedactionTest extends SAITester
+{
+    @Before
+    public void setup()
+    {
+        requireNetwork();
+    }
+
+    private static final String KEYSPACE = "prepared_stmt_cleanup";
+    private static final String createKsStatement = "CREATE KEYSPACE " + KEYSPACE +
+                                                    " WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 };";
+
+    @Test
+    public void testQuery()
+    {
+        String selectCQL2 = "SELECT * FROM t WHERE val : 'missing' OR (val : ? AND val : ?)";
+        Session session = getSession(ProtocolVersion.V5);
+        session.execute(createKsStatement);
+        session.execute("Use prepared_stmt_cleanup");
+        session.execute("CREATE TABLE t (id text PRIMARY KEY, val text)");
+        session.execute("CREATE CUSTOM INDEX ON t(val) USING 'StorageAttachedIndex' WITH OPTIONS = {" +
+                        "'index_analyzer': '{\n" +
+                        "\t\"tokenizer\":{\"name\":\"ngram\", \"args\":{\"minGramSize\":\"2\", \"maxGramSize\":\"3\"}}," +
+                        "\t\"filters\":[{\"name\":\"lowercase\"}]\n" +
+                        "}'," +
+                        "'query_analyzer': '{\n" +
+                        "\t\"tokenizer\":{\"name\":\"whitespace\"},\n" +
+                        "\t\"filters\":[{\"name\":\"porterstem\"}]\n" +
+                        "}'};");
+
+        PreparedStatement preparedSelect2 = session.prepare(selectCQL2);
+        session.execute(preparedSelect2.bind("'quick'", "'dog'")).all().size();
+    }
+}
```

Comment:
This is not a proper test, it was just to help me print while working on the patch. Proper testing will be added when we agree on the actual queue (latest vs slowest, singleton, etc)

This prints the following example for redacting in the logs:
```
TRACE [ReadStage-1] 2024-09-18 18:50:32,200 Plan.java:361 - Optimizing plan:
Limit 5000 (rows: 0.0, cost/row: 0.0, cost: 0.0..0.0)
 └─ Filter (val : <redacted> OR (val : <redacted> AND val : <redacted>)) (sel: NaN) (rows: 0.0, cost/row: 0.0, cost: 0.0..0.0)
     └─ Fetch (rows: 0.0, cost/row: 0.0, cost: 0.0..0.0)
         └─ Union (keys: 0.0, cost/key: 0.0, cost: 0.0..0.0)
             ├─ LiteralIndexScan of t_val_idx (sel: 0.000000000, step: 0.0) (keys: 0.0, cost/key: 0.0, cost: 0.0..0.0)
             │  predicate: Expression{name: val, op: MATCH, lower: <redacted>, upper: <redacted>, exclusions: []}
             └─ Intersection (keys: 0.0, cost/key: 0.0, cost: 0.0..0.0)
                 ├─ LiteralIndexScan of t_val_idx (sel: 0.000000000, step: 0.0) (keys: 0.0, cost/key: 0.0, cost: 0.0..0.0)
                 │  predicate: Expression{name: val, op: MATCH, lower: <redacted>, upper: <redacted>, exclusions: []}
                 └─ LiteralIndexScan of t_val_idx (sel: 0.000000000, step: 0.0) (keys: 0.0, cost/key: 0.0, cost: 0.0..0.0)
                    predicate: Expression{name: val, op: MATCH, lower: <redacted>, upper: <redacted>, exclusions: []}

TRACE [ReadStage-1] 2024-09-18 18:50:32,201 Plan.java:372 - Candidate query plan:
Limit 5000 (rows: 0.0, cost/row: 0.0, cost: Infinity..Infinity)
 └─ Filter (val : <redacted> OR (val : <redacted> AND val : <redacted>)) (sel: 0.000000000) (rows: 0.0, cost/row: 0.0, cost: Infinity..Infinity)
     └─ Fetch (rows: 0.0, cost/row: 0.0, cost: Infinity..Infinity)
         └─ Everything (keys: 0.0, cost/key: 0.0, cost: Infinity..Infinity)

TRACE [MemtableFlushWriter:2] 2024-09-18 18:50:32,201 LifecycleTransaction.java:239 - Committing transaction over [] staged: [obsolete: [], update: []], logged: [obsolete: [], update: []]
TRACE [ReadStage-1] 2024-09-18 18:50:32,202 Plan.java:372 - Candidate query plan:
Limit 5000 (rows: 0.0, cost/row: 0.0, cost: 0.0..0.0)
 └─ Filter (val : <redacted> OR (val : <redacted> AND val : <redacted>)) (sel: NaN) (rows: 0.0, cost/row: 0.0, cost: 0.0..0.0)
     └─ Fetch (rows: 0.0, cost/row: 0.0, cost: 0.0..0.0)
         └─ Union (keys: 0.0, cost/key: 0.0, cost: 0.0..0.0)
             ├─ LiteralIndexScan of t_val_idx (sel: 0.000000000, step: 0.0) (keys: 0.0, cost/key: 0.0, cost: 0.0..0.0)
             │  predicate: Expression{name: val, op: MATCH, lower: <redacted>, upper: <redacted>, exclusions: []}
             └─ LiteralIndexScan of t_val_idx (sel: 0.000000000, step: 0.0) (keys: 0.0, cost/key: 0.0, cost: 0.0..0.0)
                predicate: Expression{name: val, op: MATCH, lower: <redacted>, upper: <redacted>, exclusions: []}

TRACE [ReadStage-1] 2024-09-18 18:50:32,202 Plan.java:372 - Candidate query plan:
Limit 5000 (rows: 0.0, cost/row: 0.0, cost: Infinity..Infinity)
 └─ Filter (val : <redacted> OR (val : <redacted> AND val : <redacted>)) (sel: 0.000000000) (rows: 0.0, cost/row: 0.0, cost: Infinity..Infinity)
     └─ Fetch (rows: 0.0, cost/row: 0.0, cost: Infinity..Infinity)
         └─ Everything (keys: 0.0, cost/key: 0.0, cost: Infinity..Infinity)

TRACE [ReadStage-1] 2024-09-18 18:50:32,202 Plan.java:379 - Optimized plan:
Limit 5000 (rows: 0.0, cost/row: 0.0, cost: 0.0..0.0)
 └─ Filter (val : <redacted> OR (val : <redacted> AND val : <redacted>)) (sel: NaN) (rows: 0.0, cost/row: 0.0, cost: 0.0..0.0)
     └─ Fetch (rows: 0.0, cost/row: 0.0, cost: 0.0..0.0)
         └─ Union (keys: 0.0, cost/key: 0.0, cost: 0.0..0.0)
             ├─ LiteralIndexScan of t_val_idx (sel: 0.000000000, step: 0.0) (keys: 0.0, cost/key: 0.0, cost: 0.0..0.0)
             │  predicate: Expression{name: val, op: MATCH, lower: <redacted>, upper: <redacted>, exclusions: []}
             └─ LiteralIndexScan of t_val_idx (sel: 0.000000000, step: 0.0) (keys: 0.0, cost/key: 0.0, cost: 0.0..0.0)
                predicate: Expression{name: val, op: MATCH, lower: <redacted>, upper: <redacted>, exclusions: []}

TRACE [MemtableFlushWriter:2] 2024-09-18 18:50:32,202 LogTransaction.java:455 - Completing txn [nb_txn_flush_68b82e70-7610-11ef-a854-a9f24348a0d4.log in /Users/ekaterina.dimitrova/cassandra/build/test/cassandra/data/system/IndexInfo-9f5c6374d48532299a0a5094af9ad1e3] with last record COMMIT:[,0,0][2613697770]
TRACE [MemtableFlushWriter:2] 2024-09-18 18:50:32,202 LogTransaction.java:320 - Removing files for transaction [nb_txn_flush_68b82e70-7610-11ef-a854-a9f24348a0d4.log in /Users/ekaterina.dimitrova/cassandra/build/test/cassandra/data/system/IndexInfo-9f5c6374d48532299a0a5094af9ad1e3]
TRACE [ReadStage-1] 2024-09-18 18:50:32,203 QueryController.java:369 - Query execution plan:
Limit 5000 (rows: 0.0, cost/row: 0.0, cost: 0.0..0.0)
 └─ Filter (val : <redacted> OR (val : <redacted> AND val : <redacted>)) (sel: NaN) (rows: 0.0, cost/row: 0.0, cost: 0.0..0.0)
     └─ Fetch (rows: 0.0, cost/row: 0.0, cost: 0.0..0.0)
         └─ Union (keys: 0.0, cost/key: 0.0, cost: 0.0..0.0)
             ├─ LiteralIndexScan of t_val_idx (sel: 0.000000000, step: 0.0) (keys: 0.0, cost/key: 0.0, cost: 0.0..0.0)
             │  predicate: Expression{name: val, op: MATCH, lower: <redacted>, upper: <redacted>, exclusions: []}
             └─ LiteralIndexScan of t_val_idx (sel: 0.000000000, step: 0.0) (keys: 0.0, cost/key: 0.0, cost: 0.0..0.0)
                predicate: Expression{name: val, op: MATCH, lower: <redacted>, upper: <redacted>, exclusions: []}
```